mount/mount.cvmfs.cc

```diff
@@ -68,21 +68,34 @@ static string MkFqrn(const string &repository) {
   return repository;
 }
 
+static bool IsFuseTInstalled() {
+  string fuseTComponentsPaths[] = { "/usr/local/bin/go-nfsv4", 
+                                          "/usr/local/lib/libfuse-t.dylib",
+                                           "/usr/local/lib/libfuse-t.a" };
+  const int pathsNumber = sizeof(fuseTComponentsPaths) / sizeof(fuseTComponentsPaths[0]);
+  platform_stat64 info;
+  for (int idx = 0; idx < pathsNumber; ++idx) {
+    bzero(&info, sizeof(platform_stat64));
+    int retval = platform_stat(fuseTComponentsPaths[idx].c_str(), &info);
+    if ((retval != 0) || !S_ISREG(info.st_mode)) {
+      return false;
+    }
+  }
+  return true;
+}
 
 static bool CheckFuse() {
-  string fuse_device;
-  int retval;
 #ifdef __APPLE__
-  retval = system("/Library/Filesystems/macfuse.fs/Contents/Resources/"
-                  "load_macfuse");
-  if (retval != 0) {
-    LogCvmfs(kLogCvmfs, kLogStderr, "Failed loading macFUSE");
-    return false;
+  bool is_fuse_t_installed = IsFuseTInstalled();
+  if (!is_fuse_t_installed) {
+    LogCvmfs(kLogCvmfs, kLogStderr, "FUSE-T installation check failed"); 
   }
-  fuse_device = "/dev/macfuse0";
+  return is_fuse_t_installed;
 #else
   fuse_device = "/dev/fuse";
 #endif
+  string fuse_device;
```

Comment:
```suggestion
  string fuse_device;
```
For Linux, I think you need to move this declaration back up